lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -179,15 +179,22 @@ def create_launch_config(connection, module):
     instance_type = module.params.get('instance_type')
     spot_price = module.params.get('spot_price')
     instance_monitoring = module.params.get('instance_monitoring')
-    assign_public_ip = module.params.get('assign_public_ip')
     kernel_id = module.params.get('kernel_id')
     ramdisk_id = module.params.get('ramdisk_id')
     instance_profile_name = module.params.get('instance_profile_name')
     ebs_optimized = module.params.get('ebs_optimized')
     classic_link_vpc_id = module.params.get('classic_link_vpc_id')
     classic_link_vpc_security_groups = module.params.get('classic_link_vpc_security_groups')
     bdm = BlockDeviceMapping()
-
+    
+    if module.params.get('assign_public_ip'):
+        if module.params.get('assign_public_ip') == 'yes':
+            assign_public_ip = True
+        else:
+            assign_public_ip = False
+    else:
+        assign_public_ip = None
```

Comment:
The `bool` type handles more cases than just `yes`. To assign the public_ip variable here to be true/false when provided or None by default, all you should need is to set the default to `None` as below, then assign the variable using:

```
module.params.get('assign_public_ip')
```